File: frontend/src/utils/pricing.js
// Single source of truth for the "Today Only 10% OFF" promotional
// pricing on the frontend - mirrors backend/app/utils/pricing.py so the
// pre-purchase preview always agrees with what the server will charge.
export const DISCOUNT_RATE = 0.1;
export const GST_RATE = 0.18;

function round2(value) {
  return Math.round((value + Number.EPSILON) * 100) / 100;
}

export function calculateDiscountAmount(price) {
  return round2(Number(price) * DISCOUNT_RATE);
}

export function calculateDiscountedPrice(price) {
  return round2(Number(price) - calculateDiscountAmount(price));
}

export function calculateGst(amount) {
  return round2(Number(amount) * GST_RATE);
}

export function calculateGrandTotal(amount) {
  return round2(Number(amount) + calculateGst(amount));
}

File: backend/app/utils/pricing.py
"""Single source of truth for the 'Today Only 10% OFF' promotional
pricing: discount and GST math, shared by PurchaseService and
LoanService instead of each re-deriving it."""
from decimal import Decimal

from app.utils.money import round2

DISCOUNT_RATE = Decimal("0.10")
GST_RATE = Decimal("0.18")


def calculate_discount_amount(price) -> Decimal:
    return round2(Decimal(price) * DISCOUNT_RATE)


def calculate_discounted_price(price) -> Decimal:
    return round2(Decimal(price) - calculate_discount_amount(price))


def calculate_gst(amount) -> Decimal:
    return round2(Decimal(amount) * GST_RATE)


def calculate_grand_total(amount) -> Decimal:
    return round2(Decimal(amount) + calculate_gst(amount))
